controllers/reconcile.go

```diff
@@ -578,6 +581,21 @@ func (r *MicroK8sControlPlaneReconciler) scaleDownControlPlane(ctx context.Conte
 	node := deleteMachine.Status.NodeRef
 
 	logger = logger.WithValues("machineName", deleteMachine.Name, "nodeName", node.Name)
+
+	logger.Info("deleting node from dqlite", "machineName", deleteMachine.Name, "nodeName", node.Name)
+
+	// NOTE(Hue): We do this step as a best effort since this whole logic is implemented to prevent a not-yet-reported bug.
+	// If we have more than 2 machines left, get cluster agent client and delete node from dqlite
+	if len(machines) > 1 {
```

Comment:
This is 2 and more machines right? Did we test what happens when we try to remove a node from a 2-machine cluster? Do we get any HA/raft issues?

Reply:
I tested with 2 local lxc containers. After the two nodes were joined, deployed an NGINX with 2 replicas. Deleted one of the nodes by running the `.remove` command on the `dqlite` binary. The node is not in the `.cluster` output but it's still there in the `kubectl get nodes` output (which is expected). One of the NGINX replicas which was scheduled on the deleting node is still there and running. Even after deleting the node using `kubectl delete node`, the replica is running on that node (which is still expected because we also need to run `microk8s leave` on the leaving node). 
But in the CAPI situation we're not calling `microk8s leave`, but rather we're deleting the machine. So I deleted the deleting node lxc container and the NGINX pod just got deleting and rescheduled on the remaining node. 
I guess it's safe to say we won't have an issue here. 
My main question is that how were we able to get along with _not removing the entry from dqlite_ until now. IIUC at some point we should be in a spot were the outdated/stale entries in dqlite are more than the alive/responsive ones.